前端 一期

senkiii opened the pull request.

robinWongM committed:
chore: remove root folder

robinWongM committed:
chore: set up dev environment, remove duplicated lock file

robinWongM committed:
chore: use pascal case for SFC & vue-router for routing

robinWongM committed:
fix: avoid tracking .DS_Store

robinWongM committed:
feat: do not need vue-class-component anymore

robinWongM committed:
feat: add eslint, prettier & lint on save

senkiii committed:
add PingForm

简单完成ping功能

senkiii committed:
Fix @vue/eslint-config-prettier & lint

senkiii committed:
Merge branch 'main' into dev

robinWongM commented on net-toolbox/workspace.code-workspace:
```diff
@@ -0,0 +1,8 @@
+{
```
这个文件应该是不被提交到 Git 仓库的

robinWongM commented on src/App.vue:
```diff
@@ -1,31 +1,57 @@
+        strong
+        secondary
+        type="info"
+        onclick='window.location.href="#/ping"'
```
1. 用 vue-router 实现跳转吧
2. 一般用 Vue 提供的绑定事件指令 `v-on`，简写 `@`：https://v3.cn.vuejs.org/guide/events.html ；不用这种直接在标签上写 onclick 的

senkiii replied:
用router-link的话好像就没法使用组件库？🤔等下再改这个

robinWongM replied:
- https://www.naiveui.com/zh-CN/light/components/menu#select.vue
- 或者，https://router.vuejs.org/zh/guide/essentials/navigation.html

robinWongM commented on src/App.vue:
```diff
@@ -1,31 +1,57 @@
-      <n-button id='agent' strong secondary type='info' onclick="window.location.href='#/agent'">Agent管理</n-button>
+      <HelloUser id="hellouser" />
+      <n-button
+        id="ping"
```
这里的 id 用处是什么呢

senkiii replied:
写css用

robinWongM commented on src/components/AgentManagement.vue:
```diff
@@ -1,3 +1,5 @@
-    <h1>Agent管理</h1>
-</template>
\ No newline at end of file
+
```
比较奇怪的空行

robinWongM commented on src/components/DefaultPage.vue:
```diff
@@ -0,0 +1,29 @@
+<template text-align="center">
+  <div id="blank"></div>
```
这个只是为了用来撑大高度的么？可以用其他元素的 margin 来实现吧

robinWongM commented on src/components/FetchUserName.vue:
```diff
@@ -1,28 +1,23 @@
+      userName: 'null',
+    };
+  },
+  reDirect() {
```
`redirect`

robinWongM commented on src/components/HelloUser.vue:
```diff
@@ -1,48 +1,50 @@
 <template>
-    <h1 size=30>{{time}}好，{{username}}。</h1>
+
```
空行

robinWongM commented on src/components/HelloUser.vue:
```diff
@@ -1,48 +1,50 @@
+    },
+    getUserName() {
+      this.username = 'Admin';
+      //不知道写啥，应该是从fetchusername组件获得username，总之先占个位置在这里
```
其实没看明白那个组件的作用

senkiii replied:
其实我也没写明白，这个版本我们先当无事发生过。。

robinWongM commented on src/components/NotFound.vue:
```diff
@@ -1,29 +1,38 @@
-    </template>
-  </n-result>
+
+    <div id="blank"></div>
```
同上

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -1,25 +1,148 @@
 <template>
-    <h1>Ping查询</h1>
+  <div id="pingform">
```
命名规范：
1. HTML 中不区分大小写，所以 tag 名、属性名都用 kebab-case，id 和 class 的值也用纯小写，但是用 `-` 或者 `_` 来分割单词
2. JS 中用 camelCase，不用 - 和 _

```suggestion
  <div id="ping-form">
```

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -1,25 +1,148 @@
+    <n-button text type="info" id="selectagent">选择Agent</n-button>
+    <n-select
+      id="selector"
+      v-model:value="chosenvalue"
```
```suggestion
      v-model:value="chosenValue"
```

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -1,25 +1,148 @@
+    <n-button text type="info" id="inputurl">目标参数</n-button>
+    <n-input
+      id="to-ping"
+      v-model:value="to_ping"
```
```suggestion
      v-model:value="toPing"
```

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -1,25 +1,148 @@
+      type="text"
+      placeholder="在这里输入你要Ping的地址"
+    />
+    <n-button type="info" @click="GoPing" id="go-ping">Go!</n-button>
```
```suggestion
    <n-button type="info" @click="goPing" id="go-ping">Go!</n-button>
```

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -1,25 +1,148 @@
+    NDataTable,
+  },
+  methods: {
+    GoPing() {
```
```suggestion
    goPing() {
```

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -1,25 +1,148 @@
+});
+</script>
+<style>
+#pingform {
```
```suggestion
#ping-form {
```

robinWongM commented on src/main.ts:
```diff
@@ -1,9 +1,13 @@
-import NotFound from './components/NotFound.vue'
-import AgentManagement from './components/AgentManagement.vue'
-import {createRouter, createWebHashHistory} from 'vue-router'
+
```
空行

robinWongM commented on vue.config.js:
```diff
@@ -1,4 +1,6 @@
-const { defineConfig } = require('@vue/cli-service')
+
```
空行

robinWongM reviewed (changes requested):
Review 完成，请看下评论

robinWongM commented:
之后开 Pull Request 的时候，标题和详情简要说一下变动的内容吧，自动生成的这个 `Dev` 就。。。没啥用

senkiii committed:
Update GoPing()->goPing()

Co-authored-by: Robin Wong <chaowang0313@gmail.com>

senkiii committed:
Update GoPing()->goPing()

Co-authored-by: Robin Wong <chaowang0313@gmail.com>

senkiii committed:
Update div id

senkiii committed:
修改代码格式

删空行。。。怎么这么多空行。。
改命名规范

senkiii committed:
Merge branch 'dev' of https://github.com/SYSU-ECNC/net-toolbox-frontend into dev

robinWongM commented on .gitignore:
```diff
@@ -18,3 +18,4 @@ pnpm-debug.log*
 *.njsproj
 *.sln
 *.sw?
+net-toolbox/workspace.code-workspace
```
已经被 Git 跟踪过的文件需要 `git rm --cached` 才可以移除跟踪

senkiii committed:
Update menu button

NaiveUI的menu组件可以和router-link配合，但是感觉不太好用。。改为把NButton渲染为<a>，虽然还是没用上routerlink，但是总比在按钮里写js好一点（

senkiii committed:
Update Ping setting & datatable

- 我的英文水平不足以支撑我给commit写总结
- 增加了“发包数量”表单
- 修改了“管理测试”表格的显示方式
- 应该不会再追踪workspace.code-workspace了

senkiii committed:
搭建Agent管理页面+修改ping页面表格格式

ping页面的“管理测试”表格现在有两个bug：
- 点击下拉菜单的时候会把所有行都展开
- 从别的页面跳转回来时下拉按钮消失
之后再改

senkiii committed:
增加了重置Token按钮

robinWongM commented on src/components/AgentManagement.vue:
```diff
@@ -1,3 +1,84 @@
+
+import { NButton, NDataTable } from 'naive-ui';
+
+const createColumns = () => {
+  return [
+    {
+      title: 'Agent name',
+      key: 'name',
+    },
+    {
+      title: '在线情况',
+      key: 'is_online',
+    },
+    {
+      title: '操作',
+      key: 'action',
+    },
+  ];
+};
+
+const agentColumns = createColumns();
```
其实这里为什么要搞成一个函数呢

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -11,50 +11,132 @@
+  {
+    time: '2022.2.2 12:22',
+    agents: 'M5',
+    isDone: 'true',
```
应该是 `true` 而不是 `'true'`

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -11,50 +11,132 @@
+];
+
+function tableDataProcess() {
+  //之后从后端获取JSON，在这里加一句JSON.parse……话说在JSON里写'true'的话，转换过来应该也还是字符串吧？
```
JSON 本身是可以表达布尔值的，而不是用字符串去套起来，所以应该直接拿到 `true` 和 `false`

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -11,50 +11,132 @@
+    } else {
+      tableData[i].isDone = '进行中';
+    }
+  }
```
一般不推荐把获取到的数据进行二次修改（immutable）。如果是为了让这个 `true` 和 `false` 在页面上以别的文字来显示，可以将这个转换逻辑写在 template 里（比如 `{{ isDone ? '已完成' : '进行中' }}`），或者用 computed/method 动态转换。

senkiii committed:
修复了PingForm点击下拉按钮后所有行都展开的bug

但没完全修复。。如果两条ping记录的时间相同就还是会有问题

robinWongM commented on src/components/PingForm.vue:
```diff
@@ -33,6 +33,9 @@
       :columns="tableColumns"
       :data="tableData"
       :max-height="250"
+      :row-key="(tableData: any) => {
+        return `${tableData.time}`;
+      }"
```
或许这样就可以了？
```suggestion
      :row-key="(tableData: any) => tableData.time"
```

> 虽然我觉得 key 应当是 id 之类的，而不是时间
> 尽量别用 `any`

senkiii committed:
增加getUserName接口

senkiii committed:
简单修改getUserName

senkiii committed:
完成agent选择器

senkiii committed:
agent查询适配接口

senkiii committed:
ping提交&ping结果查询功能

senkiii committed:
获取token

senkiii committed:
删除agent的UI（有bug）

不知道为什么引入的NInput没有办法使用了……

senkiii committed:
bug修好了

上一个commit中忘记把组件引入components了……

senkiii committed:
获取token&删除agent
